search/searcher/search_phrase_test.go

```diff
@@ -159,6 +159,71 @@ func TestMultiPhraseSearch(t *testing.T) {
 	}
 }
 
+func TestSloppyMultiPhraseSearch(t *testing.T) {
+	soptions := search.SearcherOptions{
+		SimilarityForField: func(field string) search.Similarity {
+			return similarity.NewBM25Similarity()
+		},
+		Explain:            true,
+		IncludeTermVectors: true,
+	}
+
+	tests := []struct {
+		phrase [][]string
+		docids []uint64
+		slop   int
+	}{
+		{
+			phrase: [][]string{{"angst"}, {"beer"}, {"database"}},
+			docids: []uint64{},
+			slop:   0,
+		},
+		{
+			phrase: [][]string{{"angst"}, {"beer"}, {"database"}},
+			docids: []uint64{baseTestIndexReaderDirect.docNumByID("2")},
+			slop:   1,
+		},
+		{
+			phrase: [][]string{{"apple", "angst"}, {"dank"}},
+			docids: []uint64{baseTestIndexReaderDirect.docNumByID("3")},
+			slop:   2,
+		},
```

Comment:
I think it would be helpful to test/illustrate how terms in the wrong order behave with respect to the sloppyness parameter.  So, perhaps a search for `beer angst` at various sloppy levels, as users might wonder if that will require sloppyness of 1 or 2 (and without reviewing the previous code I'm not sure, but guess it's 2).

Reply:
I am not sure either, I'll add a test case in the second commit